src/middlewares/mail.js

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable import/prefer-default-export */
+import { verifyEmail } from '../services/mail/template/verifyEmail';
+import { emailVerifyToken } from '../utils/index';
+import Mail from '../services/mail/Mail';
+
+export const SendVerificationEmail = async (req, res, next) => {
+  let { email, firstName, lastName } = req.body;
+  email = email ? email.trim() : '';
+  firstName = firstName.trim();
+  lastName = lastName.trim();
+  const id = 'some_encoded_identiity';
+  const token = await emailVerifyToken(id);
+  const emaildDetails = {
+    Subject: 'Email Verification',
+    Recipient: email,
+  };
+  const link = `http://localhost:3000/api/v1/auth/verify?id=${token}`;
+  const data = {
+    email,
+    firstName,
+    lastName,
+    link,
+  };
+
+  try {
+    const send = new Mail(emaildDetails, verifyEmail(data));
```

Comment:
@zanio the `verifyEmail` function should be better named to denote what it's actually doing which is to generate a `verify email template` hence rename to `verifyEmailTemplate`

Reply:
Fixed